src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -118,12 +151,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         userEntity.setProperty(USER_FRIENDS_LIST_PROPERTY, Arrays.asList(friends));
       }
     }
+    getAndStoreBlobKeys(request, userEntity);
     datastore.put(userEntity);
 
     // Redirect to the profile page, and let the front-end know the current logged in user
     response.sendRedirect("/profile.html?id=" + userId);
   }
 
+  /** Method that stores the blob-keys (in Datastore) of files uploaded to Blobstore */
+  private void getAndStoreBlobKeys(HttpServletRequest request, Entity userEntity) {
+    // Get values determining whether or not to update/save blobkeys of any particular image
+    boolean profilePhotoUploaded = getBooleanParameter(request, USER_PROFILE_PHOTO, IMAGE_NOT_FOUND);
```

Comment:
nit: I would just call this USER_PHOTO_1 for consistency; the fact that USER_PHOTO_1 is used as the profile photo isn't relevant when it comes to storage